sycl/include/CL/sycl/ONEAPI/intel_matrix/matrix-amx.hpp

```diff
@@ -0,0 +1,439 @@
+namespace matrix {
+using namespace cl::sycl;
+using namespace cl::sycl::ONEAPI;
+
+#ifdef __SYCL_DEVICE_ONLY__
+SYCL_EXTERNAL extern "C" _tile1024i
+_tileloadd64_internal(short row, short col, char *buf, size_t stride);
+SYCL_EXTERNAL extern "C" _tile1024i
+_tdpbssd_internal(unsigned short m, unsigned short n, unsigned short k,
+                  _tile1024i dst, _tile1024i src1, _tile1024i src2);
+SYCL_EXTERNAL extern "C" _tile1024i
+_tdpbf16ps_internal(unsigned short m, unsigned short n, unsigned short k,
+                    _tile1024i dst, _tile1024i src1, _tile1024i src2);
+SYCL_EXTERNAL extern "C" void _tilestored64_internal(short row, short col,
+                                                     char *buf, size_t stride,
+                                                     _tile1024i tile);
+static _tile1024i tileloadd64_internal(short row, short col, char *buf,
+                                       size_t stride) {
+  return _tileloadd64_internal(row, col, buf, stride);
+}
+static _tile1024i tdpbssd_internal(unsigned short m, unsigned short n,
+                                   unsigned short k, _tile1024i dst,
+                                   _tile1024i src1, _tile1024i src2) {
+  return _tdpbssd_internal(m, n, k, dst, src1, src2);
+}
+static _tile1024i tdpbf16ps_internal(unsigned short m, unsigned short n,
+                                     unsigned short k, _tile1024i dst,
+                                     _tile1024i src1, _tile1024i src2) {
+  return _tdpbf16ps_internal(m, n, k, dst, src1, src2);
+}
+static void tilestored64_internal(short row, short col, char *buf,
+                                  size_t stride, _tile1024i tile) {
+  return _tilestored64_internal(row, col, buf, stride, tile);
+}
+#endif
+
+#ifndef __SYCL_DEVICE_ONLY__
+static _tile1024i tileloadd64_internal(short row, short col, char *buf,
+                                       size_t stride) {
+  return __builtin_ia32_tileloadd64_internal(row, col, buf, stride);
+}
+static _tile1024i tdpbssd_internal(unsigned short m, unsigned short n,
+                                   unsigned short k, _tile1024i dst,
+                                   _tile1024i src1, _tile1024i src2) {
+  return __builtin_ia32_tdpbssd_internal(m, n, k, dst, src1, src2);
+}
+static _tile1024i tdpbf16ps_internal(unsigned short m, unsigned short n,
+                                     unsigned short k, _tile1024i dst,
+                                     _tile1024i src1, _tile1024i src2) {
+  return __builtin_ia32_tdpbf16ps_internal(m, n, k, dst, src1, src2);
+}
+static void tilestored64_internal(short row, short col, char *buf,
+                                  size_t stride, _tile1024i tile) {
+  __builtin_ia32_tilestored64_internal(row, col, buf, stride, tile);
+}
+#endif
+
+enum class matrix_layout { row_major, col_major, packed_a, packed_b };
+
+#define TILE_SZ 16
+
+template <typename Group, typename T, size_t NUM_ROWS = detail::dynamic_extent,
```

Comment:
```suggestion
template <typename Group, typename T, size_t NumRows = detail::dynamic_extent,
```
Usually ALL_CAPS is recommended for macro names only.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-not-CAPS
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es32-use-all_caps-for-all-macro-names

LLVM coding style suggests using camel case - https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly.

Please, apply to all other parameters.